x-pack/plugins/security_solution/public/alerts/components/alerts_table/actions.tsx

```diff
@@ -125,17 +132,24 @@ export const sendAlertToTimelineAction = async ({
         const { timeline } = formatTimelineResultToModel(timelineTemplate, true);
         const query = replaceTemplateFieldFromQuery(
           timeline.kqlQuery?.filterQuery?.kuery?.expression ?? '',
-          ecsData
+          ecsData,
+          timeline.timelineType
         );
         const filters = replaceTemplateFieldFromMatchFilters(timeline.filters ?? [], ecsData);
         const dataProviders = replaceTemplateFieldFromDataProviders(
           timeline.dataProviders ?? [],
-          ecsData
+          ecsData,
+          timeline.timelineType
         );
+
         createTimeline({
           from,
           timeline: {
             ...timeline,
+            timelineType: TimelineType.default,
+            templateTimelineId: timelineId,
```

Comment:
would this be ```resultingTimeline.templateTimelineId ?? null```

Reply:
why do we care about templateTimelineId?

Reply:
it is confusing to say that `timelineType` is default but we need to pass templateTimelineId